File: OOP V1/4_single_inheritance.cpp

#include<bits/stdc++.h>
using namespace std;

class Animal{
    public:
    string name;
    void eat(){
        cout<< name <<" Can eat"<<endl;
    }
};

class Cat:public Animal{
    public:
    Cat(string name){
        this->name = name;
    }
};
                    
int main(){
    Cat obj("Cat");
    obj.eat(); 
            
              
    return 0;
}



//                      Public Derivation         Private D            Protected D

// Public Member            Public             Private             Protected  

// Private Member          Not inherited      Not Inherited       Not inherited

// Protected Member        Protected           Private               Protected

File: OOP V1/4_single_inheritance.py
class Animal:
    def __init__(self,name) -> None:
        self.name = name
    def eat(self):
        print(self.name, "Can Eat")
            
class Cat(Animal):
    def __init__(self, name) -> None:
        # super().__init__(name)
        Animal.__init__(self,name)

obj = Cat("Cat")
obj.eat()

